### PR TITLE
FOGL-41 adding template file and license.txt

### PR DESCRIPTION
**Pull request by praveen-garg**
[FOGL-41](https://scaledb.atlassian.net/browse/FOGL-41)

**TODO:** 

- [ ] ~$~**{**{FOGLAMP_LICENSE_DESCRIPTION}**}** in `__template__.py`
- [ ] setup.py  Need author_email  `?@osisoft.com`
- [ ] docs/conf.py

**author and version:** both can be set as template variable Or need to be put manually as needed.

`__author__ = ${FULL_NAME}`

`__version__ = ${VERSION}` **Not sure if we need this?!**

### CODE REVIEW

**Comment by Terris-ScaleDB** [comment]
Please add Mark as reviewer. I am not familiar with ${ } .. what replaces those?

File: __template__.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    * You have to also use ``sphinx.ext.todo`` extension
+
+.. _Google Python Style Guide:
+   http://google.github.io/styleguide/pyguide.html
+
+"""
+
+import sys
+
+__author__ = ${FULL_NAME}
```
Comment (MarkRiddoch) [inline __template__.py]: Was your  thought to add a git checkin filter that substitutes these in first checkin?

File: __template__.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
+# -*- coding: utf-8 -*-
+"""
+${FOGLAMP_LICENSE_DESCRIPTION}
+
+Homepage and documentation: http://foglamp.readthedocs.io/
```
Comment (MarkRiddoch) [inline __template__.py]: I would be tempted to rephrase this as

See: http://fogloamp.readthedocs.io/

This gives us some flexibility if we want to make a separate homepage for the product

File: __template__.py (hunk 1 above)
Comment (praveen-garg) [inline __template__.py]: no; I think I was not able to explain in PR description.

PyCharm allows File and Code Templates; so we can set these placeholders with template variables and have snippets to insert.

> author and version: both can be set as template variable 
> 
> __author__ = ${FULL_NAME}
> __version__ = ${VERSION}
> 

Or need to be put (replace) manually

**Comment by praveen-garg** [comment]
@MarkRiddoch @izoratti @billehunt `{{FOGLAMP_LICENSE_DESCRIPTION}}` still need to be placed in this.

**Comment by Terris-ScaleDB** [comment]
I don't think we're going to be able to specify the exact boilerplate now. The boilerplate's first and last lines should contain tags so a script can later change them en masse.  For example:

"""
BEGIN
(c) 2017 ..

END
"""

I would not actually use BEGIN and END.

**Comment by Terris-ScaleDB** [comment]
Will the boilerplate appear in the generated docstring documentation for the module? That would not be ideal.

**Comment by Terris-ScaleDB** [comment]
When a new is created, what is the process for updating the version strings en masse?

**Comment by praveen-garg** [comment]
Fixed in https://github.com/foglamp/FogLAMP/pull/35
